tests/js/searchtools.js

```diff
@@ -1,16 +1,19 @@
 describe('Basic html theme search', function() {
 
+  function loadFixture(name) {
+      req = new XMLHttpRequest();
+      req.open("GET", `base/tests/js/fixtures/${name}`, false);
+      req.send(null);
+      return req.responseText;
+  }
+
   describe('terms search', function() {
 
     it('should find "C++" when in index', function() {
-      index = {
-        docnames:["index"],
-        filenames:["index.rst"],
-        terms:{'c++':0},
-        titles:["&lt;no title&gt;"],
-        titleterms:{}
-      }
-      Search.setIndex(index);
+      searchindex = loadFixture("cpp/searchindex.js");
+      eval(searchindex);
```

Comment:
You could perhaps create another function called `evalFixture` for that? or is there a more canonical way to do it with karma?

Reply:
I'll double-check the documentation; perhaps there is a better way (it would seem like importing a script file per-test could be a common use case).

Reply:
I didn't find anything suitable in `karma` itself.  I did find some plugins.. there's [a reasonably popular one](https://github.com/billtrik/karma-fixture) that loads HTML and JSON, but I don't think it is keen on [evaluating JavaScript](https://github.com/billtrik/karma-fixture/issues/19).

I had a look at `jasmine`, too.  We could enable a plugin to provide [mocking of AJAX requests](https://jasmine.github.io/tutorials/mocking_ajax) -- but again given that the code is running in a browser context, then we need to read the fixture data from somewhere.  I'll think about this some more.